components/Radio/src/Radio.cpp

```diff
@@ -11,58 +11,109 @@
 
 #include "Radio.h"
 
+#include "freertos/queue.h"
 #include <driver/uart.h>
 #include <esp_log.h>
+#include <string.h>
+
+
+
+#define FTDI_PORT UART_NUM_0
+#define BUF_SIZE (1024)
+#define RD_BUF_SIZE (BUF_SIZE)
 
 namespace radio {
 
 #if RAD_ENABLED == true
 
 static const char* TAG = "UART";
+static QueueHandle_t uart0_queue;
 
 Radio::Radio()
     : Task("UART", 4096, 5)
 {
 }
 
-void Radio::run(void* data)
+void Radio::init(callbackFunction fn) 
 {
-    ESP_LOGD(TAG, "UART start");
-
+   //The callback handler represent the function passed from any module that implement 
+   //serial communication and want to receive serial through callback
+   callback_handler = fn;
+   
+    ESP_LOGD(TAG, "Initializing UART");
     uart_config_t uart_config = {
         .baud_rate = 115200,
         .data_bits = UART_DATA_8_BITS,
         .parity = UART_PARITY_DISABLE,
         .stop_bits = UART_STOP_BITS_1,
         .flow_ctrl = UART_HW_FLOWCTRL_DISABLE,
-        .rx_flow_ctrl_thresh = 122,
+        .rx_flow_ctrl_thresh = 127,
         .use_ref_tick = false,
     };
 
     ESP_LOGD(TAG, "Configuring UART parameters");
-    ESP_ERROR_CHECK(uart_param_config(UART_NUM_2, &uart_config));
+    ESP_ERROR_CHECK(uart_param_config(UART_NUM_0, &uart_config));
 
     ESP_LOGD(TAG, "Configuring UART pins");
-    ESP_ERROR_CHECK(uart_set_pin(UART_NUM_2, RAD_TX_PIN, RAD_RX_PIN, UART_PIN_NO_CHANGE, UART_PIN_NO_CHANGE));
-
-    const int uart_buffer_size = (1024 * 2);
-    QueueHandle_t uart_queue;
+   //ESP_ERROR_CHECK(uart_set_pin(UART_NUM_0, RAD_TX_PIN, RAD_RX_PIN, UART_PIN_NO_CHANGE, UART_PIN_NO_CHANGE));
 
     ESP_LOGD(TAG, "Installing UART driver");
-    ESP_ERROR_CHECK(uart_driver_install(UART_NUM_2, uart_buffer_size, uart_buffer_size, 10, &uart_queue, 0));
-
-    while (true) {
-        ESP_LOGD(TAG, "Reading bytes");
-
-        std::uint8_t buffer[13];
-        int n = uart_read_bytes(UART_NUM_2, buffer, 12, 1000);
-        buffer[12] = '\0';
-        if (n == 12) {
-            ESP_LOGI(TAG, "Received, buffer = %s", reinterpret_cast<char*>(buffer));
-        } else {
-            ESP_LOGE(TAG, "Error, n = %d", n);
+    ESP_ERROR_CHECK(uart_driver_install(UART_NUM_0, BUF_SIZE * 2, BUF_SIZE * 2, 10, &uart0_queue, 0));
+
+}
+
+void Radio::sendDataToRadio(void* data, size_t length)
+{
+    uart_write_bytes(FTDI_PORT, (const char*) data, length);
+}
+
+void Radio::run(void* data)
+{   
+    uart_event_t event;
+    size_t buffered_size;
+    uint8_t* dtmp = (uint8_t*) malloc(RD_BUF_SIZE);
```

Comment:
This doesn't need to be allocated with `malloc`, it can live on the stack.

Reply:
```suggestion
    uint8_t dtmp[RD_ BUF_SIZE];
```

Reason: the size is constant, does not change.